cli/src/main/java/com/devonfw/tools/ide/cli/Ide.java

```diff
@@ -94,8 +109,113 @@ public int run(String... args) {
+        initializeJlineCompletion();
+        return ProcessResult.SUCCESS;
+      } catch (RuntimeException e) {
+        return 1;
+      }
+    }
+
     CliArgument first = CliArgument.of(args);
     CliArgument current = initContext(first);
+    return processCliArgument(current);
+  }
+
+  /**
+   * Initializes jline3 autocompletion.
+   */
+  private void initializeJlineCompletion() {
+
+    try {
+      ContextCommandlet init = new ContextCommandlet();
+      init.run();
+      this.context = init.getIdeContext();
+
+      // Supplier<Path> workDir = context::getCwd;
+      // set up JLine built-in commands
+      // TODO: fix BuiltIns or remove
+      // Builtins builtins = new Builtins(workDir, null, null);
+      // builtins.rename(Builtins.Command.TTOP, "top");
+      // builtins.alias("zle", "widget");
+      // builtins.alias("bindkey", "keymap");
+
+      Parser parser = new DefaultParser();
+      try (Terminal terminal = TerminalBuilder.builder().build()) {
+
+        // initialize our own completer here
+        IdeCompleter completer = new IdeCompleter(init, context);
+
+        LineReader reader = LineReaderBuilder.builder().terminal(terminal).completer(completer).parser(parser)
+            .variable(LineReader.LIST_MAX, AUTOCOMPLETER_MAX_RESULTS).build();
+
+        // Create autosuggestion widgets
+        AutosuggestionWidgets autosuggestionWidgets = new AutosuggestionWidgets(reader);
+        // Enable autosuggestions
+        autosuggestionWidgets.enable();
+
+        // TODO: implement TailTipWidgets
+        // TailTipWidgets widgets = new TailTipWidgets(reader, systemRegistry::commandDescription, 5,
+        // TailTipWidgets.TipType.COMPLETER);
+        // widgets.enable();
+        // TODO: add own KeyMap
+        // KeyMap<Binding> keyMap = reader.getKeyMaps().get("main");
+        // keyMap.bind(new Reference("tailtip-toggle"), KeyMap.alt("s"));
+
+        String prompt = "prompt> ";
+        String rightPrompt = null;
+        String line;
+
+        AnsiConsole.systemInstall();
+        while (true) {
+          try {
+            line = reader.readLine(prompt, rightPrompt, (MaskingCallback) null, null);
+            reader.getHistory().add(line);
+            try {
+              runCommand(line, init);
+              init.resetRunParams();
+            } catch (Exception e) {
+              context.error("An error occurred while running the CLI command:{} {}", line, e);
+            }
+          } catch (UserInterruptException e) {
+            // Ignore CTRL+C
+          } catch (EndOfFileException e) {
+            // CTRL+D
+            return;
+          } catch (Exception e) {
+            context.error("An error occurred while using autocompletion: {}", e);
+          } finally {
+            AnsiConsole.systemUninstall();
+          }
+        }
+
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    } catch (Throwable e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
What is the point in doing this?
A `Throwable` can be an `OutOfMemoryError` and creating a `RuntimeException` with another `StackTrace` may in the worst case again end in an out of memory causing a JVM crash.
If you want to handle checked exceptions and wrap them as `RuntimeExecption` find but this outer try - catch block seems pointless to me. If you have something like that catching `Exception` will still be sufficient and if you do generic exception wrapping after all, you can use the chance to enrich the exception message with contextual information ("Unexpected error during interactive auto-completion").

Reply:
Changed Throwable to Exception now.